### ISSUE TITLE
Docs/terminology for expected revision

### ISSUE THREAD

**Issue by wiegell** [EventStore/EventStore#4287]
I have met all of these terms in the docs/source:
- Expected revision
- Prepare position
- Commit position
- Stream revision
- Next expected version
- Current version
- Stream position

It's not easy to tell them apart, since they are not explicitly documented. I suspect many of them mean the same. The revision concept is quite central to EventStore IMO, but it is difficult to get a grasp of when the documentation of it is vague. I would suggest you settle on a terminlogy, document it clearly and then make the libraries adhere to that terminology too.

**Comment by wiegell** [comment]
Great explanation, thanks alot. Some thoughts:

- Rename the places where "Current version" and "Current revision" are used to "Stream Revision"
- Rename "Prepare position" to "Commit position"
- Rename "Next expected version" to "Next expected revision"
- I forgot some terms also used "Event #" (admin gui), "Event Number" and "Event Revision" (go client). The explanation in the docs is OK, but i miss an explicit connection between "Event number" and "Stream revision" in the docs (that you can use an event number in a subsequent call as an expected revision). Also much of that information about stream revisions and how they behave on truncating is not in the client docs, but in the server docs (hidden in the "Features" section). I think a brief explanation about truncation and how it relates to stream revisions / eventnumbers is needed also in the client docs.
- I would suggest settling on either "Event Number" or "Event Revision".

**Comment by yordis** [comment]
Would it make sense to adopt https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/ETag ... sorry to add to the mess 😓

**Comment by alexeyzimarev** [comment]
> Rename "Prepare position" to "Commit position"

We can't do that because prepare position still exists in legacy databases, otherwise we would have already removed any notion of it.

"Event number" sounds correct. I think the terminology used in the [Navigator app](https://learn.eventstore.com/event-store-navigator-preview) is correct, but we are open for suggestions for the upcoming Preview 2 of the app.

"Event revision" doesn't make sense as events are immutable and have no revisions. "Stream revision", on the other hand, is legit. Appending an event to a stream increases the stream revision by one.

**Comment by alexeyzimarev** [comment]
@yordis eTags are a bit different:

> Entity tag that uniquely represents the requested resource. It is a string of [ASCII](https://developer.mozilla.org/en-US/docs/Glossary/ASCII) characters placed between double quotes, like "675af34563dc-tr34".

The only unique identifiers of a resources in ESDB would be:
- Stream name combined with the event number. It only identifies an event.
- Stream name as the unique identifier of a stream

Stream revision cannot be an eTag or part of eTag as it doesn't identify the stream, it becomes the event eTag. Also, in most cases you aren't interested to get one event by its eTag. In most cases you either read the stream from the start or from a certain revision to the end of the stream, or read the stream backwards. Both cases can be classified as reading dynamic resources.

**Comment by yordis** [comment]
Yeah ... I was thinking only about the name, not the spec. I see multiple folks mapping the HTTP Etag directly into the expected version ... but I leave it to you, you know better. I'm looking forward to aligning on the topic!

**Comment by alexeyzimarev** [comment]
> Rename "Next expected version" to "Next expected revision"

This makes sense btw
